census/estimator/trainer/task.py

```diff
@@ -0,0 +1,180 @@
+import argparse
+
+import model
+
+import tensorflow as tf
+from tensorflow.contrib.learn.python.learn import learn_runner
+from tensorflow.contrib.learn.python.learn.utils import (
+    saved_model_export_utils)
+
+
+tf.logging.set_verbosity(tf.logging.INFO)
+
+
+def generate_experiment_fn(train_files,
+                           eval_files,
+                           num_epochs=None,
+                           train_batch_size=40,
+                           eval_batch_size=40,
+                           embedding_size=8,
+                           first_layer_size=100,
+                           num_layers=4,
+                           scale_factor=0.7,
+                           **experiment_args):
+  """Create an experiment function given hyperparameters.
+  
+  See command line help text for description of args.
+  Returns:
+    A function (output_dir) -> Experiment where output_dir is a string
+    representing the location of summaries, checkpoints, and exports.
+    this function is used by learn_runner to create an Experiment which
+    executes model code provided in the form of an Estimator and
+    input functions.
+
+    All listed arguments in the outer function are used to create an
+    Estimator, and input functions (training, evaluation, serving). 
+    Unlisted args are passed through to Experiment.
+  """
+  def _experiment_fn(output_dir):
+    # num_epochs can control duration if train_steps isn't
+    # passed to Experiment
+    train_input = model.generate_input_fn(
+        train_files,
+        num_epochs=num_epochs,
+        batch_size=train_batch_size,
+    )
+    # Don't shuffle evaluation data
+    eval_input = model.generate_input_fn(
+        eval_files,
+        batch_size=eval_batch_size,
+        shuffle=False
+    )
+    return tf.contrib.learn.Experiment(
+        model.build_estimator(
+            job_dir,
+            embedding_size=embedding_size,
+            # Construct layers sizes with exponetial decay
+            hidden_units=[
+                max(2, int(first_layer_size * scale_factor**i))
+                for i in range(num_layers)
+            ]
+        ),
+        train_input_fn=train_input,
+        eval_input_fn=eval_input,
+        # export strategies control the prediction graph structure
+        # of exported binaries.
+        export_strategies=[saved_model_export_utils.make_export_strategy(
```

Comment:
Totally agree. In particular, we need to transform the output class back into a string using the `tf.gather` from the `LABELS` classes aren't very useful to something like a webapp. This can easily  be done with a custom estimator. And IMO all the `Classifiers` should take a `FeatureColumn` for `label` that automatically does this transformation, as well as the work being done in `parse_label`.